assets/models/Map.js

```diff
@@ -48,17 +48,18 @@ class Map {
 
   createLayerGroups(hazards, markerParams = {}) {
     hazards?.forEach((hazard, idx) => {
-      const category = hazard?.hazard?.name?.toLowerCase();
-      const subCategory = hazard?.hazardCategory?.name?.toLowerCase();
+      const categoryId = hazard?.hazardCategory?.id;
+      const subCategoryId = hazard?.hazard?.id;
```

Comment:
Saving `categoryId` and `subCategoryId` inside marker Object for filtering purpose